src/components/Posts.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { fetchPosts } from '../actions/postActions';
+
+class Posts extends Component {
+    componentWillMount() {
+        this.props.fetchPosts();
+    }
+
+    componentWillReceiveProps(nextPost) {
+        if(nextPost.newPost){
+            this.props.posts.unshift(nextPost.newPost);
```

Comment:
`unshift` will add the new post to the top of the `Arraylist`. if we use `map` it will add new element to the bottom of the list.